src/widget_type.h

```diff
@@ -17,6 +17,25 @@
 #include "gfx_type.h"
 #include "window_type.h"
 
+/* Range Iterator Helper. This should be somewhere else, and documented. Where? */
+template <class Iter>
```

Comment:
Isn't this just a simple version of `(std::)span` (core/span_type.hpp)?

Reply:
Perhaps, now trying to work out how to use that.

Reply:
Unless I was trying it wrong, our span only seems to support iterators that are pointers.